src/uct/base/uct_mem.c

```diff
@@ -249,8 +258,9 @@ ucs_status_t uct_mem_alloc(size_t length, const uct_alloc_method_t *methods,
 
 #ifdef SHM_HUGETLB
             /* Allocate huge pages */
-            address = (flags & UCT_MD_MEM_FLAG_FIXED) ?
-                      params->address : NULL;
+            alloc_length = length;
+            address      = (flags & UCT_MD_MEM_FLAG_FIXED) ?
```

Comment:
UCT_MEM_ALLOC_PARAM_FIELD_ADDRESS ?

Reply:
Haven't really change the logic here. This is how it looks in 1.9.x
```
#ifdef SHM_HUGETLB
            /* Allocate huge pages */
            alloc_length = min_length;
            address = (flags & UCT_MD_MEM_FLAG_FIXED) ? addr : NULL;
            status = ucs_sysv_alloc(&alloc_length, min_length * 2, &address,
                                    SHM_HUGETLB, alloc_name, &shmid);
            if (status == UCS_OK) {
                goto allocated_without_md;
            }
#else
```

Seems like sysv_alloc requires FLAG_FIXED to be passed. Should we change the logic here?

Reply:
ok